src/test/ui/consts/const_discriminant.rs

```diff
@@ -0,0 +1,37 @@
+// run-pass
+#![feature(const_discriminant)]
+#![allow(dead_code)]
+
+use std::mem::{discriminant, Discriminant};
+
+#[inline(never)]
+fn identity<T>(x: T) -> T { x }
```

Comment:
Maybe add a comment explaining why we added this function? It can be quite confusing otherwise.^^